### PR TITLE
Bugfix/798-z-index in calendar

### PR DESCRIPTION
**Pull request by ckoutsiaris**
Closes sap/fundamental https://github.com/SAP/fundamental/issues/798

I believe we should review the usage of z-indexes. At the moment is minimal but possibly we can remove some of them. We also need to agree on the restrictions that we might have to put for more complex components. Eg Are we going to allow popups inside modal?

#### Test

* http://localhost:4000/
* http://localhost:3030/

#### Changelog

z-indexes map. Options: base,first,second,top,default

screenshot 2018-10-26 at 14 45 32

### CODE REVIEW

**Comment by joseegm** [comment]
Test Version for this PR was deployed

Built with commit d0ad99f01250d56810937cc7f5039b3a2476777e

https://deploy-preview-831--fundamental.netlify.com

File: scss/core/forms.scss [hunk 1]
```diff
@@ -72,7 +72,7 @@ select,
         border-width: 0 0 2px 2px;
         transform: rotate(-45deg);
         position: absolute;
-        z-index: 2;
+        z-index: map-get($fd-z-index-levels, "second");
```
Comment (ianquigley-sap) [inline scss/core/forms.scss]: Still not sure why this forms element needs a z-index?  Is it to be displayed on top of something else?

Comment (ckoutsiaris) [reply scss/core/forms.scss]: Yeah, maybe we can remove it. @saadmhybris what do you think?

Comment (eboyer) [reply scss/core/forms.scss]: @ianquigley-sap @ckoutsiaris @saadmhybris ~ I believe that is the little dropdrown arrow that should be positioned on top of the select.

File: scss/components/input-group.scss [hunk 2]
```diff
@@ -74,7 +74,7 @@ $block: #{$fd-namespace}-input-group;
         }
 
 
-        z-index: 2; // Brings focus state above input
+        z-index: map-get($fd-z-index-levels, "first"); // Brings focus state above input
```
Comment (eboyer) [inline scss/components/input-group.scss]: @ckoutsiaris ~ This one looks like it should be "second"

**Comment by eboyer** [comment]
@ckoutsiaris ~ Really like the thinking behind these z-indexs. Agreed that we could think about them even more, this is an awesome start!

File: scss/components/input-group.scss (hunk 2 above)
Comment (ckoutsiaris) [reply scss/components/input-group.scss]: @eboyer This line actually fixes the bug. I am not sure if it breaks something though :( From a quick visual checking, I didn't notice anything.

**Pull request merged**